### PR TITLE
Add ability to set bimodal pathfinding

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Signed-off-by: Nitesh Balusu <niteshbalusu@icloud.com>

Fixes: #1149

### CODE REVIEW

**Comment by hsjoberg** [comment]
Can we rework the actual db item to be`bimodal`/`appriori` instead of `true`/`false`?

**Comment by niteshbalusu11** [comment]
> Can we rework the actual db item to be`bimodal`/`appriori` instead of `true`/`false`? 

Done.

File: src/storage/app.ts [hunk 1]
```diff
@@ -227,6 +229,7 @@ export const setupApp = async () => {
     setItemObject<boolean>(StorageItem.rescanWallet, false),
     setItemObject<boolean>(StorageItem.receiveViaP2TR, false),
     setItemObject<boolean>(StorageItem.strictGraphPruningEnabled, false),
+    setItemObject<string>(StorageItem.lndPathfindingAlgorithm, DEFAULT_PATHFINDING_ALGORITHM),
```
Comment (hsjoberg) [inline src/storage/app.ts:232]: Change to `setItem` instead as it's a `string`

File: src/state/Settings.ts [hunk 2]
```diff
@@ -166,6 +171,7 @@ export const settings: ISettingsModel = {
     actions.setRescanWallet(await getRescanWallet());
     actions.setReceiveViaP2TR(await getItemObject(StorageItem.receiveViaP2TR) ?? false);
     actions.setStrictGraphPruningEnabled(await getItemObject(StorageItem.strictGraphPruningEnabled) ?? false);
+    actions.setLndPathfindingAlgorithm(await getItemObject(StorageItem.lndPathfindingAlgorithm) ?? false);
```
Comment (hsjoberg) [inline src/state/Settings.ts:174]: change to `setItem` and `?? DEFAULT_PATHFINDING_ALGORITHM`

**Review by hsjoberg — CHANGES_REQUESTED** [review]

File: src/windows/Settings/Settings.tsx [hunk 3]
```diff
@@ -1094,6 +1094,16 @@ ${t("experimental.tor.disabled.msg2")}`;
     toast(t("msg.written", { ns:namespaces.common }));
   };
 
+  // Bimodal path finding
+  const lndPathfindingAlgorithm = useStoreState((store) => store.settings.lndPathfindingAlgorithm);
+  const changeBimodalPathFindingEnabled = useStoreActions((store) => store.settings.changeLndPathfindingAlgorithm);
+  const changeBimodalPathFindingEnabledPress = async () => {
+    const modal = lndPathfindingAlgorithm === ('apriori' || null) ? 'bimodal' : 'apriori';
```
Comment (hsjoberg) [inline src/windows/Settings/Settings.tsx:1101]: I think this line is incorrect
It should be 

```suggestion
    const modal = (lndPathfindingAlgorithm === 'apriori' || lndPathfindingAlgorithm === null) ? 'bimodal' : 'apriori';
```

**Comment by niteshbalusu11** [comment]
Fixed.

**Review by hsjoberg — APPROVED** [review]

**Review by hsjoberg — APPROVED** [review]

**Comment by hsjoberg** [comment]
LGTM!

**Pull request merged**